launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -359,8 +359,9 @@ void FirstLaunchView::extractGogData()
 			return; // should not happen - but avoid deleting wrong folder in any case
 
 		QString tmpFileExe = tempDir.filePath("h3_gog.exe");
+		QString tmpFileBin = tempDir.filePath("h3_gog-1.bin");
 		QFile(fileExe).copy(tmpFileExe);
-		QFile(fileBin).copy(tempDir.filePath("h3_gog-1.bin"));
+		QFile(fileBin).copy(tmpFileBin);
```

Comment:
Unrelated to this PR, but could be this copy somehow causing those "file corruption" errors?

Never heard about corrupted downloads via browsers for like 10+ years. It is only gog installers that are corrupted for some reason.

What if original installer in Downloads or where user has it is fine, but Qt fails to copy it for some reason?

What if we were to change this code to only copy in case of "bad" filenames, as in - bin name is not `stem(exe) + "-1.bin`?

Reply:
Not sure. But if it would be a 'random' copy error: It wouldn't be reproducible multiple times.

The idea was to copy the files from outside the sandbox to inside to avoid any access problems when working further with the files.

Maybe the new information (hashes, filesize) will help us to find errors.

But I think most problems caused by the sandboxing (remember one case we've got hashes calculated on phone which are correct).

But in general I don't not have any experience here. But it seems to be a android only issue.

Reply:
If goal is to debug this case then we could also try to get file size / checksum of original files. If they don't match with copied files - then it is copy that is the problem.

Reply:
Good idea. Can also add this case.